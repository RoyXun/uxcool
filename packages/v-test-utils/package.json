{
  "name": "@suning/v-test-utils",
  "version": "1.0.10",
  "main": "src/index.js",
  "module": "src/index.js",
  "license": "MIT",
  "scripts": {
    "start": "npm run server",
    "compile": "v-tools run compile",
    "server": "v-tools run server"
  },
  "peerDependencies": {
    "vue": ">= 2.5.*"
  },
  "devDependencies": {
    "@suning/v-tools": "^1.0.35",
    "@vue/test-utils": "^1.0.0-beta.16"
  },
  "dependencies": {
    "@suning/v-trigger": "^1.0.64",
    "@vue/server-test-utils": "^1.0.0-beta.30",
    "jquery": "^3.3.1",
    "vue-server-renderer": "^2.6.11"
  }
}
