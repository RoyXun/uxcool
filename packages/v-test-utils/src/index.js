import $ from 'jquery';

export * from '@vue/test-utils';

export { $, $ as jQuery };
// ms
export async function waitTime(delay) {
  await new Promise((resolve) => {
    setTimeout(() => {
      resolve();
    }, delay);
  });
}
