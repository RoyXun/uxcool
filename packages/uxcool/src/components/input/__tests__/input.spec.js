import { mount } from '@suning/v-test-utils';
import {
  UxInput, UxSearchInput, UxTextarea, UxInputGroup
} from '..';

describe('ux-input', () => {
  it('render allow-clear correctly', () => {
    const wrapper = mount(UxInput, {
      propsData: {
        allowClear: true,
        value: 'test value',
      },
    });
    wrapper.find('i').trigger('click');
    expect(wrapper.vm.innerValue).toBe('');
  });

  it('show word limit correctly', () => {
    const wrapper = mount(UxInput, {
      propsData: {
        showWordLimit: true,
        maxlength: 20,
        value: 'test value',
      },
    });
    expect(wrapper.find('.ux-input-limit-word').element.innerHTML).toBe('10/20');
  });

  describe('addon and prefix/suffix', () => {
    it('with props', () => {
      /**
       * although the prefix and suffix can be use by props according to the handler program,
       * however they have not been defined in props (by 19040734)
       */
      const wrapper = mount(UxInput, {
        propsData: {
          addonBefore: 'addon-before',
          addonAfter: 'addon-after',
        },
      });
      expect(wrapper.findAll('.ux-input-group-addon').at(0).element.innerHTML).toBe('addon-before');
      expect(wrapper.findAll('.ux-input-group-addon').at(1).element.innerHTML).toBe('addon-after');
    });

    it('with slots', () => {
      const wrapper = mount(UxInput, {
        slots: {
          prefix: 'prefix',
          suffix: 'suffix',
          addonBefore: 'addon-before',
          addonAfter: 'addon-after',
        },
      });
      expect(wrapper.find('.ux-input-prefix').element.innerHTML).toBe('prefix');
      expect(wrapper.find('.ux-input-suffix').element.innerHTML).toBe('suffix');
      expect(wrapper.findAll('.ux-input-group-addon').at(0).element.innerHTML).toBe('addon-before');
      expect(wrapper.findAll('.ux-input-group-addon').at(1).element.innerHTML).toBe('addon-after');
    });
  });

  it('emit event correctly', () => {
    const pressenterFn = jest.fn();
    const keydownFn = jest.fn();
    const blurFn = jest.fn();
    const wrapper = mount(UxInput, {
      listeners: {
        pressenter: pressenterFn,
        keydown: keydownFn,
        blur: blurFn,
      },
    });
    wrapper.trigger('focusin');
    wrapper.trigger('keydown.enter');
    expect(pressenterFn).toHaveBeenCalledTimes(1);
    wrapper.trigger('blur');
    expect(blurFn).toHaveBeenCalledTimes(1);
  });
});

describe('ux-search-input', () => {
  it('render search button correctly', () => {
    const searchFn = jest.fn();
    const wrapper = mount(UxSearchInput, {
      propsData: {
        // eslint-disable-next-line
        enterButton: (h) => <span>search</span>,
      },
      listeners: {
        search: searchFn,
      },
    });
    expect(wrapper.find('button').element.innerHTML).toBe('<span>search</span>');
    wrapper.find('button').trigger('click');
    expect(searchFn).toHaveBeenCalledTimes(1);
  });
});

describe('ux-textarea', () => {
  it('render style correctly', () => {
    const wrapper = mount(UxTextarea, {
      propsData: {
        textareaStyle: {
          backgroundColor: 'black',
        },
      },
    });
    expect(wrapper.find('textarea').element.style.backgroundColor).toBe('black');
  });

  it('render auto height correctly', async () => {
    // const wrapper = mount(UxTextarea, {
    //   propsData: {
    //     autoSize: { minRow: 2, maxRow: 6 },
    //     textareaStyle: { width: '100px', fontSize: '14px', padding: '0px' },
    //   },
    // });
    // cannot be tested due to height is uncatchable in jsdom
    // TODO
  });
});

describe('test input group', () => {
  it('compact mode', () => {
    const Test = {
      render() {
        return <UxInput style="width: 50%" />;
      },
    };
    const wrapper = mount(UxInputGroup, {
      slots: {
        default: [Test, Test],
      },
      propsData: {
        compact: true,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
  it('different size', () => {
    const wrapper = mount(UxInputGroup, {
      slots: {
        default: UxInput,
      },
      propsData: {
        size: 'small',
      },
    });
    expect(wrapper.element).toMatchSnapshot();
    wrapper.setProps({ size: 'large' });
    expect(wrapper.element).toMatchSnapshot();
  });
});
