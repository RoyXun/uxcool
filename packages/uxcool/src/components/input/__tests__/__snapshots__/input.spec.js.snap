// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test input group compact mode 1`] = `
<span
  class="ux-input-group ux-input-group-compact"
>
  <input
    class="ux-input"
    style="width: 50%;"
  />
  <input
    class="ux-input"
    style="width: 50%;"
  />
</span>
`;

exports[`test input group different size 1`] = `
<span
  class="ux-input-group ux-input-group-sm"
>
  <input
    class="ux-input"
  />
</span>
`;

exports[`test input group different size 2`] = `
<span
  class="ux-input-group ux-input-group-sm"
>
  <input
    class="ux-input"
  />
</span>
`;
